src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function
+groupByKey<T, K extends keyof T>(items: T[], key: K): GroupsMap<T> {
+  const result: GroupsMap<T> = items.reduce((acc, ch) => {
+    const keyValue = ch[key] as unknown as string;
+
+    if (acc[keyValue] === undefined) {
```

Comment:
pls [see falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
```suggestion
    if (!acc[keyValue]) {
```